src/anm/utils.js

```diff
@@ -43,6 +43,41 @@ is.tween = function(f) {
     return f.is_tween && is.modifier(f);
 };
 
+is.equal = function(x, y) {
+    if (x === y) return true;
+    // if both x and y are null or undefined and exactly the same
+
+    if (!(x instanceof Object) || !(y instanceof Object)) return false;
+    // if they are not strictly equal, they both need to be Objects
+
+    if (x.constructor !== y.constructor) return false;
+    // they must have the exact same prototype chain, the closest we can do is
+    // test their constructor.
+
+    for (var p in x) {
```

Comment:
Isn't test for equality becomes more resource-wasting than finding a value between the same ones? Fortunately, most of the player classes have no deep prototype chains. I'd honestly prefer some hash-driven comparison or something like that.